ksqldb-common/src/main/java/io/confluent/ksql/parser/NodeLocation.java

```diff
@@ -18,24 +18,71 @@
 import com.google.errorprone.annotations.Immutable;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.OptionalInt;
 
 @Immutable
+@SuppressWarnings("checkstyle:CyclomaticComplexity")
 public final class NodeLocation {
 
-  private final int line;
-  private final int charPositionInLine;
+  private final int startLine;
+  private final int startCharPositionInLine;
+  private final OptionalInt startStartIndex;
+  private final OptionalInt startStopIndex;
+  private final OptionalInt endLine;
+  private final OptionalInt stopCharPositionInLine;
+  private final OptionalInt stopStartIndex;
+  private final OptionalInt stopStopIndex;
```

Comment:
should I just expose all of these internal attributes via a getter just in case people might need to access them? @wicknicks

Reply:
yeah that sounds reasonable. Some Java docs would be great too. thanks!

Reply:
nit: let's expose them via a wrapper object (e.g. `TokenLocation` which contains `line/posInLine/startIdx/StopIdx` and then just have `getStartTokenLocation` and `getEndTokenLocation`). follow up question, does ANTLR already have a class for that?

Reply:
yes, it's called `Token`. I wanted to do that, but it seemd that it wasn't a part of the classpath for module

Reply:
added wrapper `TokenLocation`